### DISCUSSION TITLE
Install dependencies that aren't in PyPI

### DISCUSSION THREAD

**Discussion by rodrigomologni** [pypa/setuptools#2913] [Q&A] [answered]
Hello!

I would like to install dependencies that aren't in PyPI. I wrote the `setup.cfg` in two ways:

```
install_requires =
    mypackage*.whl
```
```
install_requires =
    mypackage
dependency_links = git+ssh://git@github.com/.../mypackage.git
```

Both scripts produce the following error:

```
$ pip install -e .
Obtaining file:///.../my2ndpackage
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
ERROR: Could not find a version that satisfies the requirement mypackage (from my2ndpackage) (from versions: none)
ERROR: No matching distribution found for mypackage
```

However, installation via `pip` works:

```
pip install mypackage*.whl
pip install git+ssh://git@github.com/.../mypackage.git
```

**Comment by webknjaz** [comment, marked as answer]
`dependency_links` has been deprecated long ago. Distribution packages only declare the dependency specifications, not the indexes. Also, this problem can be solved on the pip level with requirements files.
Perhaps https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/ and https://caremad.io/posts/2013/07/setup-vs-requirement/ will help you understand why it has to work this way.

**Reply by rodrigomologni** [reply]
Thank you very much, @webknjaz! :-)
